src/utils/utils_part.c

```diff
@@ -184,9 +184,85 @@ void ocf_part_set_valid(struct ocf_cache *cache, ocf_part_id_t id,
 			cache->conf_meta->valid_parts_no--;
 			part->config->priority = OCF_IO_CLASS_PRIO_LOWEST;
 			part->config->min_size = 0;
-			part->config->max_size = PARTITION_SIZE_MAX;
+			part->config->max_size = cache->conf_meta->cachelines;
 			ENV_BUG_ON(env_strncpy(part->config->name, sizeof(part->config->name),
 					"Inactive", 9));
 		}
 	}
 }
+
+static inline uint32_t ocf_part_evict_size(struct ocf_request *req)
+{
+	uint32_t needed_cache_lines, part_available, cache_lines_to_evict;
+	uint32_t part_occupancy, part_occupancy_debt;
+	uint32_t part_occupancy_limit =
+		ocf_part_get_max_size(req->cache, req->part_id);
+
+	needed_cache_lines = ocf_engine_repart_count(req) +
+		ocf_engine_unmapped_count(req);
+
+	part_occupancy = ocf_part_get_occupancy(req->cache, req->part_id);
+
+	if (part_occupancy_limit >= part_occupancy) {
+		part_available = part_occupancy_limit - part_occupancy;
+		part_occupancy_debt = 0;
+	} else {
+		/* Occupancy is greater than occupancy limit. Evict missing number of
+		 * cachelines */
+		part_occupancy_debt = part_occupancy - part_occupancy_limit;
+		part_available = 0;
+	}
+
+	if (ocf_freelist_num_free(req->cache->freelist) <
+			ocf_engine_unmapped_count(req)) {
+		/* Number of cachelines to insert greater than number of free
+		 * cachelines */
+		if (part_available >= needed_cache_lines) {
+			/* Cache is full, but target's part occupancy limit is not reached
+			   */
+			ocf_req_reset_part_evict(req);
+			cache_lines_to_evict = needed_cache_lines;
+		} else {
+			/* Cache is full and target part reached it's occupancy limit */
+			ocf_req_set_part_evict(req);
```

Comment:
Why use getter/setter here (not inlined) if other `struct ocf_request` fields are accessed by pointer dereference?

Reply:
I've added set&get methods to make this function more unit-testable

Reply:
Well.. It's kinda inconsistent with other fields usages, but OK. Could you make them `inline`?